### ISSUE TITLE
Ability to contribute updates for outdated packages

### ISSUE THREAD

**Issue by christianrondeau** [chocolatey/chocolatey#678]
In short: As a developer and heavy user of Chocolatey - especially due to the ability to `choco update all` - **I'd like to contribute package updates with pull requests**, when they are not maintained, or simply not updated "fast enough".

Many packages, such as [ChromeDriver2](https://chocolatey.org/packages/ChromeDriver2), do not reference a GitHub repository with the NuGet and PowerShell source files, so it is impossible for users like me to "help" other than poking the official package maintainer.

I think it would be greatly useful to *force* package maintainers to provide a public repository, therefore allowing pull requests and potentially reducing the workload of the maintainer.

This is related to [issue #308](https://github.com/chocolatey/chocolatey/issues/308), and partially covered in [issue #342](https://github.com/chocolatey/chocolatey/issues/342)

Here are a few other ideas that could reduce this problem; let me know if you'd like me to create several issues, as they all try to solve the same problem.

* Ability to flag a package as outdated. Packages getting some kind of a flag/download treshold would be marked as "outdated in Chocolatey" both in the web site and in the command line output.
* Ability to bypass the maintainer (maybe only for packages not "owned" by the maintainer) for a specific version without taking ownership (I guess this would require a fully separate approval and peer-review process...)
* Ability to create a "branch" of any package, and allowing you to tell `choco` to use yours instead but still use the same name (so that I can point back to the original package once it is updated) - this would be a temporary workaround to avoid having to uninstall and reinstall just to get a newer version faster

What are your thoughts? Do you feel that the urge to get shiny new packages as quickly as possible as well as avoiding "stale" packages is an issue for enough users to warrant such solutions? Are there other initiatives in Chocolatey that I missed?

Oh, and I have to say it: I just love Chocolatey, and can't imagine living without it. Congrats on the Kickstarter!

**Comment by ferventcoder** [comment]
Just a quick mention. Some of these are already covered as
issues/enhancements over at https://github.com/chocolatey/chocolatey.org.
Would you mind subscribing/commenting on the existing issues and creating
issues for the enhancements you can't find?

And we can close this ticket as wrong repository.

On Tuesday, February 17, 2015, Christian Rondeau <notifications@github.com>
wrote:

> As a developer and heavy user of Chocolatey - especially due to the
> ability to choco update all - I'd like to contribute to upgrading
> packages that are not maintained, or simply not updated "fast enough".
>
> Many packages, such as ChromeDriver2
> <https://chocolatey.org/packages/ChromeDriver2>, do not reference a
> GitHub repository with the NuGet and PowerShell source files, so it is
> impossible for users like me to "help" other than poking the official
> package maintainer.
>
> I think it would be greatly useful to *force* package maintainers to
> provide a public repository, therefore allowing pull requests and
> potentially reducing the workload of the maintainer.
>
> This is related to issue #308
> <https://github.com/chocolatey/chocolatey/issues/308>, and partially
> covered in issue #342
> <https://github.com/chocolatey/chocolatey/issues/342>
>
> Here are a few other ideas that could reduce this problem; let me know if
> you'd like me to create several issues, as they all try to solve the same
> problem.
>
>    - Ability to flag a package as outdated. Packages getting some kind of
>    a flag/download treshold would be marked as "outdated in Chocolatey" both
>    in the web site and in the command line output.
>    - Ability to bypass the maintainer (maybe only for packages not
>    "owned" by the maintainer) for a specific version without taking ownership
>    (I guess this would require a fully separate approval and peer-review
>    process...)
>    - Ability to create a "branch" of any package, and allowing you to
>    tell choco to use yours instead but still use the same name (so that I
>    can point back to the original package once it is updated) - this would be
>    a temporary workaround to avoid having to uninstall and reinstall just to
>    get a newer version faster
>
> What are your thoughts? Do you feel that the urge to get shiny new
> packages as quickly as possible as well as avoiding "stale" packages is an
> issue for enough users to warrant such solutions? Are there other
> initiatives in Chocolatey that I missed?
>
> Oh, and I have to say it: I just love Chocolatey, and can't imagine living
> without it. Congrats on the Kickstarter!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chocolatey/chocolatey/issues/678>.
>


-- 

Rob
"Be passionate in all you do"

http://devlicio.us/blogs/rob_reynolds
http://ferventcoder.com
http://twitter.com/ferventcoder

**Comment by christianrondeau** [comment]
You are correct, most of my comments apply to chocolatey.org. The closest issue I found was: https://github.com/chocolatey/chocolatey.org/issues/12 though it specifically address the visibility of outdated packages, not their resolution.

I do believe that it may apply to `choco.exe` too, but only if it exists in chocolatey.org so I'll just close this one for now.

It has been moved as-is here: https://github.com/chocolatey/chocolatey.org/issues/161

Thanks!
